src/Http/Controllers/MetadataController.php

```diff
@@ -19,7 +19,15 @@ public function index()
             \Barryvdh\Debugbar\Facade::disable();
         }
 
-        $cert = Storage::disk('samlidp')->get(config('samlidp.certname', 'cert.pem'));
+        $cert = config('samlidp.cert') ?: Storage::disk('samlidp')->get(config('samlidp.certname', 'cert.pem'));
+
+        if (strpos($cert, 'file://') === 0) {
+            if (!is_file($cert)) {
+                throw new \InvalidArgumentException(sprintf("File not found '%s'", $cert));
+            }
```

Comment:
I'm not opposed to the if statement, but this could clear it a bit up.
```suggestion
            throw_if(! is_file($cert), new \InvalidArgumentException(sprintf("File not found '%s'", $cert)));
```

Otherwise adding a space after negation for codestyle `if (! is_file($cert)) {`